File: MonotonicArray.py
class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        isUp = True
        isDown = True
        for i in range( 1, len(A) ):
            if isUp == True or isDown == True:
                if A[i] > A[i-1]:
                    isDown = False
                elif A[i] < A[i-1]:
                    isUp = False
            else:
                break
                
        if isUp == False and isDown == False:
            return False
        
        return True

File: MonotonicArray.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& A) {
        bool isUp = true, isDown = true;
        for( int i = 1; i < A.size(); ++i )
        {
            if( isUp == true || isDown == true )
            {
                if( A[i] > A[i-1] )
                {
                    isDown = false;
                }
                else if( A[i] < A[i-1] )
                {
                    isUp = false;
                }
            }
            else
            {
                break;
            }
        }
        if( isUp == false && isDown == false )
        {
            return false;
        }
        return true;
    }
};
